### PR TITLE
fix: added validation for empty reset password field

### LINKED ISSUE #10463: Missing Disabled State for 'Reset Password' Button on Empty Input
Found a bug? Please fill out the sections below. 👍

**Issue Summary**

The "Reset Password" page does not have the button disabled when there is no input.
If the user clicks the "Reset Password" button without providing any input, it will get stuck in an infinite loop.

### **Steps to Reproduce**

1. Go to Login Page
2. select Forgot? option
3. eneter your email id

### **Expected Results**

The button should be disabled when there is no input.

https://github.com/calcom/cal.com/assets/110765105/0a7e5fbb-a799-4c04-9888-282c7fd9d52e

### PR DESCRIPTION
**Pull request by VARUN949**
## What does this PR do?

I made changes to the code in apps/web/pages/auth/forgot-password/[id].tsx to address the problem.

Fixes #10463 

https://github.com/calcom/cal.com/assets/110765105/a367065a-5416-4669-a639-191dfe9bfbd4

## Requirement/Documentation

none
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Navigate to password reset page
Enter email to receive reset password Link
Open password reset mail and click on "Change Password"
Try to click on Reset Password without entering any input in password input field

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/auth/forgot-password/[id].tsx [hunk 1]
```diff
@@ -74,6 +76,14 @@ export default function Page({ requestId, isRequestExpired, csrfToken }: Props)
...
     );
   };
+  useEffect(() => {
+    // disbale Reset Password button if password input field is empty
+    if (passwordValue) {
+      setIsEmpty(passwordValue.length === 0);
+      return;
+    }
+    setIsEmpty(true);
+  }, [passwordValue]);
```
Comment (Udit-takkar) [inline apps/web/pages/auth/forgot-password/[id].tsx:86]: why are using setState and useEffect here ?
just use 

```suggestion
const isEmpty = passwordValue.length === 0
```

File: apps/web/pages/auth/forgot-password/[id].tsx [hunk 2]
```diff
@@ -24,6 +24,8 @@ export default function Page({ requestId, isRequestExpired, csrfToken }: Props)
   const formMethods = useForm<{ new_password: string }>();
   const success = formMethods.formState.isSubmitSuccessful;
   const loading = formMethods.formState.isSubmitting;
+  const [isEmpty, setIsEmpty] = useState<boolean>(true);
```
Comment (Udit-takkar) [inline apps/web/pages/auth/forgot-password/[id].tsx:27]:
```suggestion
```

Comment (VARUN949) [reply apps/web/pages/auth/forgot-password/[id].tsx]: @Udit-takkar Thank you for reviewing, made suggested changes

**Review by Udit-takkar — APPROVED** [review]

**Review by emrysal — APPROVED** [review]
looks good

**Pull request merged**